PROPOSED_FEATURES.md

```diff
@@ -0,0 +1,287 @@
+whose content may not be localized. This is useful for wrapping child elements
+which provide extended functionality in translated content.
+
+```html
+<div data-l10n-id="key1">
+  <menulist data-l10n-opaque="true"/>
+</div>
+```
+
+```properties
+key1 = Always <menulist/> when opening a new tab. 
+```
+
+
+## 7. Retain the identity of source elements
+
+As a developer, I want to store references to child elements and attach behaviors to
+them without the localization layer breaking them.
+
+```html
+<div data-l10n-id="submit-form">
+  <button id="submit">Submit</button>
+</div>
+```
+
+```properties
+submit-form = <button>Submit</button>
+```
+
+Even after multiple retranslations, the identity of the `<button>` should
+be the same. The element should not be recreated by the localization layer.
+
+```js
+let button = document.getElementById("submit");
+button.addEventListener(...);
+
+// Someplace else in the code...
+// Button is still a reference to the button in the DOM.
+button.style = "...";
+```
+
+Proposed solution: Child elements are re-used rather than recreated.
+
+References:
+* [Firefox Preferences JS code which currently cannot be cached](https://searchfox.org/mozilla-central/rev/2a6f3dde00801374d3b2a704232de54a132af389/browser/components/preferences/in-content/sync.js#148)
+* [Another Firefox Preferences JS code which currently cannot be cached](https://searchfox.org/mozilla-central/rev/2a6f3dde00801374d3b2a704232de54a132af389/browser/components/preferences/in-content/findInPage.js#35)
+
+
+## 8. Allow nested localizable elements
+
+As a developer, I want to be able to build nested structures of localizable
+elements, each with its own `data-l10n-id`.
+
+```html
+<div data-l10n-id="key1">
+  <menulist>
+    <menuitem data-l10n-id="menuitem1"/>
+    <menuitem data-l10n-id="menuitem2"/>
+    <menuitem data-l10n-id="menuitem3"/>
+  </menulist>
+</div>
+```
+
+```properties
+key1 = Always <menulist/> when opening a new tab.
+menuitem1 = sing
+menuitem2 = blink
+menuitem3 = play a sound 
+```
+
+## 9. Allow asymmetrically nested translations
+
+As a localizer, I want to be able to add more nested children than are present
+in the source.
+
+As a developer, I don't want to think about all possible permutations of nested
+elements which may appear in translations.
+
+```html
+<div data-l10n-id="key1" data-l10n-whitelist="ul, ul > li"></div>
+```
+
+Translation 1:
+```properties
+key1 = This is a list of errors with
+  <ul>
+    <li>Bullet 1</li>
+    <li>Bullet 2</li>
+    <li>Bullet 3</li>
+  </ul>
+```
+
+Translation 2:
+```properties
+key1 = This is a list of errors with
+  <ul>
+    <li>Bullet 1</li>
+    <li>Bullet 2</li>
+    <li>Bullet 3</li>
+    <li>Bullet 4</li>
+  </ul>
+```
+
+Translation 3:
+```properties
+key1 = This is a list of errors with
+  <ul>
+    <li>Bullet 1</li>
+    <li>Bullet 2</li>
+    <li>Bullet 3</li>
+  </ul>
+  <ul>
+    <li>Bullet 4</li>
+  </ul>
+```
+
+
+## 10. Allow void elements
```

Comment:
This has been dropped as of https://github.com/zbraniecki/fluent-domoverlays-js/issues/4